### ISSUE TITLE
Products forgotten in the replenishment list

### ISSUE THREAD

**Issue by E-Concept-Applications** [Dolibarr/dolibarr#26352]
### Bug

Hello,

I've noticed a big problem in the list of products to be replenished.
There are obviously counting and listing errors.

Here is my replenishment screen:
![image](https://github.com/Dolibarr/dolibarr/assets/5550794/01ecc388-789f-48b3-a839-4191947a3ea1)
I'd just like to point out that I don't see any difference if I check the box allowing me to include negative stocks if the minimum stock is not defined.

In fact, after checking the stock manually in Dolibarr, it appears that :
- There are only 7 references to replenish, if we take virtual stock into account, and not 8 as indicated in the total (circled in green).
- Only 5 references are listed.

Here are the stock details (from the product page) for the two references missing from my list:
![image](https://github.com/Dolibarr/dolibarr/assets/5550794/cdebcfeb-18ef-4f85-a508-f40bee14010c)
and:
![image](https://github.com/Dolibarr/dolibarr/assets/5550794/8476ab75-9b41-4be2-865e-c520298616a9)

Incidentally, you're supposed to be able to see products below the alert stock level in the accounting section, but that doesn't work either:
![image](https://github.com/Dolibarr/dolibarr/assets/5550794/fa3281d0-9552-4b9a-9a7e-f3f7b2d6eabc)

I should also point out that I manually checked all the real and virtual stock values to make sure there were no inconsistencies in the DB.

In short, this makes it particularly difficult to manage replenishments and stock correctly...

Thanks in advance!

David

PS: Sorry if the English is a bit rough, it's all automatic translation with Deepl...

### Environment Version

18.0.2

### Environment OS

Ubuntu Server 20.04.1

### Environment Web server

Apache 2.4

### Environment PHP

8.1.24

### Environment Database

MariaDB 10.3.38

### Environment URL(s)

/product/stock/replenish.php

**Comment by Humml87** [comment]
Hello,

i think your Problem is the same Problem/Bug what we have. 
The current code remove some replenishment after output the counter. In some other situations we have replenishment lines with "0" to buy... The fix is included there PR #23316.
My English is also not enough to explain the problem apparently, so the PR is not merged...

Best regards
Christian
